opentelemetry-otlp/src/exporter/http/logs.rs

```diff
@@ -3,13 +3,18 @@ use std::sync::Arc;
 use async_trait::async_trait;
 use http::{header::CONTENT_TYPE, Method};
 use opentelemetry::logs::{LogError, LogResult};
-use opentelemetry_sdk::export::logs::{LogData, LogExporter};
+use opentelemetry_sdk::{
+    export::logs::{LogData, LogExporter},
+    suppression::SuppressionGuard,
+};
+//use opentelemetry_sdk::suppression::Supression;
 
 use super::OtlpHttpClient;
 
 #[async_trait]
 impl LogExporter for OtlpHttpClient {
     async fn export(&mut self, batch: Vec<LogData>) -> LogResult<()> {
+        let _guard = SuppressionGuard::new();
```

Comment:
hm holding this type of guard across await points does not generally work as expected. You would likely need to do a similar future wrapping strategy that we have in https://github.com/open-telemetry/opentelemetry-rust/blob/28d8cf9b6f6fd0f7037d7499e22d35d24f51fdcc/opentelemetry/src/trace/context.rs#L375-L384

Reply:
Thanks for the suggestion @jtescher. I tried the approach you suggested, basically holding the guard within Context, and try to propagate it across async functions. Seems I am still doing something different, as it is not working. Though I will debug if further, just in case you find some issue in the approach, or any other suggestions. I initially thought it would be straightforward  and tokio `task_local` value is propagated automatically through the chain of async method calls, but probably this is not :)